src/EpisodeIntegrationService/ReceiveData/ReceiveData.cs

```diff
@@ -162,6 +164,35 @@ private EpisodeDto MapEpisodeToEpisodeDto(BssEpisode episode)
         };
     }
 
+    private DateOnly? ParseNullableDate(string? date)
+    {
+        if (string.IsNullOrEmpty(date)) return null;
+
+        if (DateTime.TryParseExact(date, AllowedDateFormats, CultureInfo.InvariantCulture, DateTimeStyles.None, out var dateTime))
+        {
+            return DateOnly.FromDateTime(dateTime);
+        }
+
+        _logger.LogError("Invalid date format: {Date}. Expected formats: {AllowedDateFormats}", date, string.Join(", ", AllowedDateFormats));
+        return null;
+    }
+
+    private DateTime? ParseNullableDateTime(string? dateTime, string format)
+    {
+        if (string.IsNullOrEmpty(dateTime)) return null;
+
+        try
+        {
+            return DateTime.ParseExact(dateTime, format, CultureInfo.InvariantCulture);
+        }
+        catch (FormatException ex)
+        {
+            _logger.LogError(ex, "Invalid date-time format: {DateTime}. Expected format: {Format}", dateTime, format);
+            return null;
+        }
+    }
```

Comment:
I think we actually want these two helper methods to throw and not catch exceptions when the date can't be parsed (the exceptions would be caught in ProcessEpisodeDataAsync). If a row has an invalid date we don't want to set it to null, we want to not process that row.